Add custom logger example

Tyharo1 opened the pull request:
The current value description for the custom logger option is not easily understandable. Adding an example to this would be greatly beneficial.

<!--
Thank you for submitting a pull request to Ajv.

Before continuing, please read the guidelines:
https://github.com/epoberezkin/ajv/blob/master/CONTRIBUTING.md#pull-requests

If the pull request contains code please make sure there is an issue that we agreed to resolve (if it is a documentation improvement there is no need for an issue).

Please answer the questions below.
-->

**What issue does this pull request resolve?**
The readability and clarity of using a custom logger when using the README.

**What changes did you make?**
Added a Javascript example of what a custom logger would like for those looking to use it.

**Is there anything that requires more attention while reviewing?**
N/A

Tyharo1 committed:
Add custom logger example

The current value description for the custom logger option is not easily understandable. Adding an example to this would be greatly beneficial.

epoberezkin commented on README.md:
````diff
@@ -1151,6 +1151,15 @@ Defaults:
 - _logger_: sets the logging method. Default is the global `console` object that should have methods `log`, `warn` and `error`. Option values:
   - custom logger - it should have methods `log`, `warn` and `error`. If any of these methods is missing an exception will be thrown.
+    ```javascript
+    var ajv = new AJV.default({
````
Not consistent with other examples, it should be `new Ajv(...`

epoberezkin commented on README.md:
````diff
@@ -1151,6 +1151,15 @@ Defaults:
+    ```javascript
+    var ajv = new AJV.default({
+      logger: {
+        log: log => console.log(log),
````
All (or most) examples in the spec atm are ES5. While it’s worth revisiting, it should stay consistent, so let’s stick to ES5 please. You can create another issue to discuss moving examples to ES6 (with the comment the Ajv code is still ES5 for wider compatibility).

epoberezkin commented on README.md:
```diff
@@ -1151,6 +1151,15 @@ Defaults:
+    var ajv = new AJV.default({
+      logger: {
+        log: log => console.log(log),
+        warn: () => null,
```
warn method is mandatory as docs say.

epoberezkin reviewed (commented):
I’ve added the comments, but in general this is a contrived example that simply creates the same behaviour as default, so it is pointless and confusing.

I suggest instead move into errors section, creating a separate subsection Error logging linked from here, and put some explanations there how any logging package can be used as long as it or its wrapper expose the required methods. Does it make sense?

Tyharo1 committed:
Adjusting Example

- Converting example to ES5
- Updating example to conform to other examples
- Updating example to have custom warning message

Tyharo1 committed:
Merge branch 'master' into patch-1

Tyharo1 committed:
Move Logging Sample To Error Section

Updated and moved logging sample to new Error Logging subsection

Tyharo1 committed:
Merge branch 'master' into patch-1

Tyharo1 committed:
Merge branch 'master' into patch-1
